openjdk15: broken build

**Describe the bug**
The derivation does not build
A clear and concise description of what the bug is.

**To Reproduce**
Steps to reproduce the behavior:
1. nix-build -A openjdk15

**Additional context**

```
/nix/store/f8ahd6cxx9ia4fzdiyczmhcv3c2qmd42-glibc-2.32-37-dev/include/bits/stdio2.h:38:34: note: '__builtin___sprintf_chk' output between 12 and 32 bytes into a destination of size 18
   38 |   return __builtin___sprintf_chk (__s, __USE_FORTIFY_LEVEL - 1,
      |          ~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   39 |       __bos (__s), __fmt, __va_arg_pack ());
      |       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Compiling 218 files for BUILD_jdk.javadoc.interim
Creating support/modules_libs/java.base/server/libjvm.so from 1015 file(s)
Compiling 31 files for BUILD_JRTFS
Creating support/modules_libs/java.base/jrt-fs.jar
Compiling 185 files for BUILD_TOOLS_JDK
Compiling 3 files for BUILD_VM_COMPILER_MATCH_PROCESSOR
Failed to create server process: Cannot run program "/nix/store/ycp10r1kk3kki07sg9wxr8qiqbcbvad9-adoptopenjdk-hotspot-bin-14.0.2/bin/java": error=0, Failed to exec spawn helper: pid: 410315, exit value: 1
IOException caught during compilation: java.io.IOException: Cannot run program "/nix/store/ycp10r1kk3kki07sg9wxr8qiqbcbvad9-adoptopenjdk-hotspot-bin-14.0.2/bin/java": error=0, Failed to exec spawn helper: pid: 410315, exit value: 1
make[3]: *** [CompileToolsJdk.gmk:46: /tmp/nix-build-openjdk-15.0.1-ga.drv-0/jdk15u-jdk-15.0.1-ga/build/linux-x86_64-server-release/buildtools/jdk_tools_classes/_the.BUILD_TOOLS_JDK_batch] Error 1
make[2]: *** [make/Main.gmk:86: buildtools-jdk] Error 2
make[2]: *** Waiting for unfinished jobs....
Compiling 5 files for BUILD_VM_COMPILER_NODEINFO_PROCESSOR
Failed to create server process: Cannot run program "/nix/store/ycp10r1kk3kki07sg9wxr8qiqbcbvad9-adoptopenjdk-hotspot-bin-14.0.2/bin/java": error=0, Failed to exec spawn helper: pid: 410360, exit value: 1
IOException caught during compilation: java.io.IOException: Cannot run program "/nix/store/ycp10r1kk3kki07sg9wxr8qiqbcbvad9-adoptopenjdk-hotspot-bin-14.0.2/bin/java": error=0, Failed to exec spawn helper: pid: 410360, exit value: 1
make[3]: *** [CompileToolsHotspot.gmk:47: /tmp/nix-build-openjdk-15.0.1-ga.drv-0/jdk15u-jdk-15.0.1-ga/build/linux-x86_64-server-release/buildtools/jdk.vm.compiler.match.processor/_the.BUILD_VM_COMPILER_MATCH_PROCESSOR_batch] Error 1
make[3]: *** Waiting for unfinished jobs....
Failed to create server process: Cannot run program "/nix/store/ycp10r1kk3kki07sg9wxr8qiqbcbvad9-adoptopenjdk-hotspot-bin-14.0.2/bin/java": error=0, Failed to exec spawn helper: pid: 410404, exit value: 1
IOException caught during compilation: java.io.IOException: Cannot run program "/nix/store/ycp10r1kk3kki07sg9wxr8qiqbcbvad9-adoptopenjdk-hotspot-bin-14.0.2/bin/java": error=0, Failed to exec spawn helper: pid: 410404, exit value: 1
make[3]: *** [CompileToolsHotspot.gmk:63: /tmp/nix-build-openjdk-15.0.1-ga.drv-0/jdk15u-jdk-15.0.1-ga/build/linux-x86_64-server-release/buildtools/jdk.vm.compiler.nodeinfo.processor/_the.BUILD_VM_COMPILER_NODEINFO_PROCESSOR_batch] Error 1
make[2]: *** [make/Main.gmk:96: buildtools-hotspot] Error 2




ERROR: Build failed for target 'all' in configuration 'linux-x86_64-server-release' (exit code 2) 

=== Output from failing command(s) repeated here ===
* For target buildtools_jdk.vm.compiler.match.processor__the.BUILD_VM_COMPILER_MATCH_PROCESSOR_batch:
Failed to create server process: Cannot run program "/nix/store/ycp10r1kk3kki07sg9wxr8qiqbcbvad9-adoptopenjdk-hotspot-bin-14.0.2/bin/java": error=0, Failed to exec spawn helper: pid: 410360, exit value: 1
IOException caught during compilation: java.io.IOException: Cannot run program "/nix/store/ycp10r1kk3kki07sg9wxr8qiqbcbvad9-adoptopenjdk-hotspot-bin-14.0.2/bin/java": error=0, Failed to exec spawn helper: pid: 410360, exit value: 1
* For target buildtools_jdk.vm.compiler.nodeinfo.processor__the.BUILD_VM_COMPILER_NODEINFO_PROCESSOR_batch:
Failed to create server process: Cannot run program "/nix/store/ycp10r1kk3kki07sg9wxr8qiqbcbvad9-adoptopenjdk-hotspot-bin-14.0.2/bin/java": error=0, Failed to exec spawn helper: pid: 410404, exit value: 1
IOException caught during compilation: java.io.IOException: Cannot run program "/nix/store/ycp10r1kk3kki07sg9wxr8qiqbcbvad9-adoptopenjdk-hotspot-bin-14.0.2/bin/java": error=0, Failed to exec spawn helper: pid: 410404, exit value: 1
* For target buildtools_jdk_tools_classes__the.BUILD_TOOLS_JDK_batch:
Failed to create server process: Cannot run program "/nix/store/ycp10r1kk3kki07sg9wxr8qiqbcbvad9-adoptopenjdk-hotspot-bin-14.0.2/bin/java": error=0, Failed to exec spawn helper: pid: 410315, exit value: 1
IOException caught during compilation: java.io.IOException: Cannot run program "/nix/store/ycp10r1kk3kki07sg9wxr8qiqbcbvad9-adoptopenjdk-hotspot-bin-14.0.2/bin/java": error=0, Failed to exec spawn helper: pid: 410315, exit value: 1

* All command lines available in /tmp/nix-build-openjdk-15.0.1-ga.drv-0/jdk15u-jdk-15.0.1-ga/build/linux-x86_64-server-release/make-support/failure-logs.
=== End of repeated output ===

No indication of failed target found.
Hint: Try searching the build log for '] Error'.
Hint: See doc/building.html#troubleshooting for assistance.

make[1]: *** [/tmp/nix-build-openjdk-15.0.1-ga.drv-0/jdk15u-jdk-15.0.1-ga/make/Init.gmk:315: main] Error 2
make: *** [/tmp/nix-build-openjdk-15.0.1-ga.drv-0/jdk15u-jdk-15.0.1-ga/make/Init.gmk:186: all] Error 2
builder for '/nix/store/5yphifxpw90h98gjkaxfqwj2p2vjxhqc-openjdk-15.0.1-ga.drv' failed with exit code 2
error: build of '/nix/store/5yphifxpw90h98gjkaxfqwj2p2vjxhqc-openjdk-15.0.1-ga.drv' failed
```

**Notify maintainers**
<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

**Metadata**
 - system: `"x86_64-linux"`
 - host os: `Linux 5.11.0-051100-generic, Ubuntu, 20.04.2 LTS (Focal Fossa)`
 - multi-user?: `no`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.3.10`
 - channels(tsteinbach): `"home-manager, nixpkgs-21.05pre270709.6b1057b452c"`
 - nixpkgs: `/home/tsteinbach/.nix-defexpr/channels/nixpkgs`

--- a/pkgs/development/compilers/adoptopenjdk-bin/jdk-linux-base.nix
+++ b/pkgs/development/compilers/adoptopenjdk-bin/jdk-linux-base.nix
@@ -82,7 +82,10 @@ let result = stdenv.mkDerivation rec {
     EOF
 
     # We cannot use -exec since wrapProgram is a function but not a command.
-    for bin in $( find "$out" -executable -type f ); do
+    #
+    # jspawnhelper is executed from JVM, so it doesn't need to wrap it, and it
+    # breaks building OpenJDK (#114495).
+    for bin in $( find "$out" -executable -type f -not -name jspawnhelper ); do
       if patchelf --print-interpreter "$bin" &> /dev/null; then
         wrapProgram "$bin" --prefix LD_LIBRARY_PATH : "${runtimeLibraryPath}"
       fi

--- a/pkgs/development/compilers/graalvm/community-edition.nix
+++ b/pkgs/development/compilers/graalvm/community-edition.nix
@@ -189,7 +189,10 @@ let
         preFixup = ''
           # We cannot use -exec since wrapProgram is a function but not a
           # command.
-          for bin in $( find "$out" -executable -type f -not -path '*/languages/ruby/lib/gems/*' ); do
+          #
+          # jspawnhelper is executed from JVM, so it doesn't need to wrap it,
+          # and it breaks building OpenJDK (#114495).
+          for bin in $( find "$out" -executable -type f -not -path '*/languages/ruby/lib/gems/*' -not -name jspawnhelper ); do
             if patchelf --print-interpreter "$bin" &> /dev/null || head -n 1 "$bin" | grep '^#!' -q; then
               wrapProgram "$bin" \
                 --prefix LD_LIBRARY_PATH : "${runtimeLibraryPath}"

--- a/pkgs/development/compilers/zulu/default.nix
+++ b/pkgs/development/compilers/zulu/default.nix
@@ -81,7 +81,10 @@ in stdenv.mkDerivation {
     EOF
   '' + lib.optionalString stdenv.isLinux ''
     # We cannot use -exec since wrapProgram is a function but not a command.
-    for bin in $( find "$out" -executable -type f ); do
+    #
+    # jspawnhelper is executed from JVM, so it doesn't need to wrap it, and it
+    # breaks building OpenJDK (#114495).
+    for bin in $( find "$out" -executable -type f -not -name jspawnhelper ); do
       wrapProgram "$bin" --prefix LD_LIBRARY_PATH : "${runtimeLibraryPath}"
     done
   '';

